f8a_report/helpers/report_helper.py

```diff
@@ -64,32 +64,113 @@ def __init__(self):
+        # Save (commit) the changes
+        self.conn.commit()
+        # Log the message returned from db cursor
+        logger.debug('%r' % self.cursor.statusmessage)
+        logger.info('Cleanup of Package Analyses data tables complete')
+
+    def cleanup_package_worker_results_data(self):
+        """Cleanup package worker results data tables on a periodic basis."""
+        # Number of days to retain the package_worker_result data
+        num_days_package_worker_result = os.environ.get
+        ('KEEP_PACKAGE_WORKER_RESULT_NUM_DAYS', '30')
+        # query to delete the package_worker_result data
+        query = sql.SQL('DELETE FROM package_worker_results '
+                        'WHERE ended_at <= NOW() - interval \'%s day\';')
+        logger.info('Starting to clean up Package Worker Result data tables')
+        # Execute the query
+        self.cursor.execute(query.as_string(self.conn) % (num_days_package_worker_result))
+        # Save (commit) the changes
+        self.conn.commit()
+        # Log the message returned from db cursor
+        logger.info('%r' % self.cursor.statusmessage)
+        logger.info('Cleanup of Package Worker Result data tables complete')
+
+    def cleanup_stack_analyses_request_data(self):
+        """Cleanup stack analyses data tables on a periodic basis."""
+        # Number of days to retain the stack_analyses_request data
+        num_days_stack_analyses_request = os.environ.get
+        ('KEEP_STACK_ANALYSES_REQUESTS_NUM_DAYS', '180')
+        # query to delete the stack_analyses_request data
+        query = sql.SQL('DELETE FROM stack_analyses_request '
+                        'WHERE DATE_DONE <= NOW() - interval \'%s day\';')
+        logger.info('Starting to clean up Stack Analyses tables')
+        # Execute the query
+        self.cursor.execute(query.as_string(self.conn) % (num_days_stack_analyses_request))
+        # Save (commit) the changes
+        self.conn.commit()
+        # Log the message returned from db cursor
+        logger.info('%r' % self.cursor.statusmessage)
+        logger.info('Cleanup of Stack Analyses tables complete')
+
+    def cleanup_api_requests_data(self):
+        """Cleanup api request data tables on a periodic basis."""
+        # Number of days to retain the api_requests data
+        num_days_api_requests = os.environ.get('KEEP_API_REQUESTS_NUM_DAYS', '180')
+        # query to delete the api_requests data
+        query = sql.SQL('DELETE FROM api_requests '
+                        'WHERE ended_at <= NOW() - interval \'%s day\';')
+        logger.info('Starting to clean up api requests data tables')
+        # Execute the query
+        self.cursor.execute(query.as_string(self.conn) % (num_days_api_requests))
+        # Save (commit) the changes
+        self.conn.commit()
+        # Log the message returned from db cursor
+        logger.info('%r' % self.cursor.statusmessage)
+        logger.info('Cleanup of api requests data tables complete')
```

Comment:
Isn't everything is same except the query and no.of days? Can this be further simplified?

A generic function which takes table name and no.of days to purge.?

Reply:
No everything is not completely same. @prashbnair had suggested  to use logger.debug for 1 case.